tools/netqtop.c

```diff
@@ -95,19 +95,24 @@ TRACEPOINT_PROBE(net, net_dev_start_xmit){
 }
 
 TRACEPOINT_PROBE(net, netif_receive_skb){
-    struct sk_buff* skb = (struct sk_buff*)args->skbaddr;
-    if(!name_filter(skb)){
+    struct sk_buff skb;
+
+    bpf_probe_read(&skb, sizeof(skb), args->skbaddr);
```

Comment:
Is this change necessary? We can directly directly get fields from skb pointer. What is the need to copy skb structure to the stack? Did you hit any issues?

Reply:
Actually we directly can get `queue_mapping` field from from skb pointer, but we can't directly call function `skb_rx_queue_recorded` & `skb_get_rx_queue` because bcc reports error. So two options:

- 1, re-implement `skb_rx_queue_recorded` & `skb_get_rx_queue` in this program.
- 2, copy skb and use kernel implemented function.

Solution 1 is not a hard work, but from my understanding, keeping the same API with kernel is better.

Reply:
Okay, I see. Let us use option 2 then. Re-implementing is not desirable.